backend/coins/btc/transactions/transactions.go

```diff
@@ -221,6 +223,18 @@ func (transactions *Transactions) SpendableOutputs() (map[wire.OutPoint]*Spendab
 	})
 }
 
+// GetHeaderTimestamp retrieves the header timestamp for a given transaction hash.
+func (transactions *Transactions) GetHeaderTimestamp(txHash chainhash.Hash) (*time.Time, error) {
+	transactions.synchronizer.WaitSynchronized()
+	return DBView(transactions.db, func(dbTx DBTxInterface) (*time.Time, error) {
```

Comment:
This adds one dbTx per UTXO. In transactions.SpendableOutputs() we aleady call `TxInfo()` on each UTXO, so it would be more natural to simply fetch the timestamp there.